google-cloud-storage/src/main/java/com/google/cloud/storage/multipartupload/model/UploadResponseParser.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.multipartupload.model;
+
+import com.google.api.client.http.HttpResponse;
+import com.google.api.core.BetaApi;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/** A utility class to parse {@link HttpResponse} and create a {@link UploadPartResponse}. */
+@BetaApi
+public final class UploadResponseParser {
+
+  private UploadResponseParser() {}
+
+  /**
+   * Parses an {@link HttpResponse} and returns a {@link UploadPartResponse}.
+   *
+   * @param response The {@link HttpResponse} from the upload part request.
+   * @return The parsed {@link UploadPartResponse}.
+   */
+  @BetaApi
+  public static UploadPartResponse parse(HttpResponse response) {
+    String eTag = response.getHeaders().getETag();
+    Map<String, String> hashes = extractHashesFromHeader(response);
+    return UploadPartResponse.builder()
+        .eTag(eTag)
+        .crc32c(hashes.get("crc32c"))
+        .md5(hashes.get("md5"))
+        .build();
+  }
+
+  static Map<String, String> extractHashesFromHeader(HttpResponse response) {
```

Comment:
This method needs unit tests.

especially because, the way it's written you will only ever get nothing, crc32c or md5 never both.

Reply:
Done